File: taller/JCZhang-Taller/JCZhang-Taller-Java/src/main/java/models/NavajaSuiza.java
package models;

public class NavajaSuiza extends Trabajador implements iChapista, iElectricista{
    String nombre;
    int aniosExp;
    int horasDiarias;

    public NavajaSuiza(String nombre, int aniosExp, int horasDiarias) {
        super(nombre, aniosExp, horasDiarias);
        this.nombre = nombre;
        this.aniosExp = aniosExp;
        this.horasDiarias = horasDiarias;
    }

    @Override
    void saludar() {
        System.out.println("Navaja Suiza: Hola! Me llamo " + nombre +" y soy una navaja suiza");
    }

    @Override
    public void arreglarChapa() {
        System.out.println("Navaja Suiza: Estoy arreglando la chapa");
    }

    @Override
    public void arreglarElectricidad() {
        System.out.println("Navaja Suiza: Estoy arreglando los circuitos eléctricos");
    }



    @Override
    public String toString(){
        return "Navaja Suiza (" +nombre+" con " + aniosExp +"años de experiencia trabaja "+ horasDiarias+ " al día)";
    }

}

File: taller/JCZhang-Taller/JCZhang-Taller-Kotlin/src/main/kotlin/models/NavajaSuiza.kt
package models

class NavajaSuiza(nombre: String, aniosExp: Int, horasDiarias: Int) : Trabajador(nombre, aniosExp, horasDiarias), iChapista, iElectricista{


    fun NavajaSuiza(nombre: String, aniosExp: Int, horasDiarias: Int){
        this.nombre = nombre
        this.aniosExp = aniosExp
        this.horasDiarias = horasDiarias
    }


    override fun saludar() {
        println("Navaja Suiza: Hola! Soy la navaja suiza y puedo hacer tanto de chapista como de electricista ")
    }

    override fun toString(): String {
        return "Navaja Suiza( $nombre con $aniosExp años de experiencia trabaja $horasDiarias horas al dia)"
    }

    override fun arreglarElectricidad() {
        println("Navaja Suiza: Estoy arreglando los circuitos eléctricos")
    }

    override fun arreglarChapa() {
        println("Navaja Suiza: Estoy arreglando la chapa")
    }
}